test/e2e_node/hugepages_test.go

```diff
@@ -167,12 +168,57 @@ func runHugePagesTests(f *framework.Framework) {
 		err := e2epod.WaitForPodSuccessInNamespace(f.ClientSet, verifyPod.Name, f.Namespace.Name)
 		framework.ExpectNoError(err)
 	})
+
+	ginkgo.It("should add resources for new huge page sizes on kubelet restart", func() {
+		ginkgo.By("Stopping kubelet")
+		startKubelet := stopKubelet()
```

Comment:
> @bart0sh where this function is used and how is it related to this PR?

It is used here

Reply:
Can you explain why stopKubelet function returns another function that starts Kubelet? It doesn't look very clear and logical to me to be honest.

Reply:
If you look at the func it make more sense. We have to store the name of the service somewhere, and imo. it is cleaner to use a closure like this instead of returning the name, or even storing it in a var.

If you have a better idea ill be happy to change the behavior! :)

Reply:
I'd say storing it in var is much more clear, readable and allows to check if the action succeeded:
```
kubeletServiceName := findRunningKubletServiceName()
err := killKubelet(kubeletServiceName)
...
err := startKubelet(kubeletServiceName)
```

Reply:
I guess we can do that, but then we will have to check with the folks from `sig-testing`. 

Doing that will move the logic for handling the different background services into the test, making it hard to refactor the code later. In case we start supporting other ways to run kubelet in the tests than `systemd`, we will have to change the tests too, not just the utilities.

I do think I prefer to separate the two concerns, but ill defer the decision to `sig-testing`.

cc @oomichi

Reply:
I think either me or you misunderstood something. I'm not proposing to support other ways to run kubelet. My proposal is to get rid of closure.
Here is how would killKubelet functions would look like:

```
func killKubelet(kubeletServiceName string) error {
	stdout, err := exec.Command("sudo", "systemctl", "kill", kubeletServiceName).CombinedOutput()
         framework.ExpectNoError(err, "Failed to kill kubelet with systemctl: %v, %v", err, stdout)
         return err
}

```

Reply:
> I'm not proposing to support other ways to run kubelet. My proposal is to get rid of closure.

Adding this code to the test will imo. move the logic of handlig the liveness of kubelet into the test, instead of a central & reusable place.
```go
kubeletServiceName := findRunningKubletServiceName()
err := killKubelet(kubeletServiceName)
...
err := startKubelet(kubeletServiceName)
````

I do agree that `no closure  > closure`, but I am not that happy with moving the systemd-logic into the test. But again, ill defer to sig-testing.

Long term the best solution would be to expose these `cmd`s into the `e2e_node/utils`, but that is outside the scope of this PR.
https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/services/kubelet.go#L183-L210

Reply:
I'm still not sure I fully understand the concern, i.e. why this code doesn't expose handling liveness of kubelet into the test:
```
startKubelet := killKubelet()
...
startKubelet()
```

but this one does:
```
kubeletServiceName := findRunningKubletServiceName()
err := killKubelet(kubeletServiceName)
...
err := startKubelet(kubeletServiceName)
```

However, I still think the latter is much more readable and understandable than the former.